### PR TITLE
Valid BST

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**
https://leetcode.com/problems/validate-binary-search-tree/

### CODE REVIEW

File: bst.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+var recursion = function(root, min, max) {
+    if (!root) return true;
+    return root.val < max && root.val > min
+        ? recursion(root.left, min, root.val) && recursion(root.right, root.val, max)
+        : false;
+};
+
+var isValidBST = function(root) {
+    if (!root || (!root.left && !root.right)) return true;
```
Comment (DzyubSpirit) [inline bst.js]: This will work the same
```suggestion
    if (!root) return true;
```

File: bst.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var recursion = function(root, min, max) {
+    if (!root) return true;
+    return root.val < max && root.val > min
+        ? recursion(root.left, min, root.val) && recursion(root.right, root.val, max)
+        : false;
+};
+
+var isValidBST = function(root) {
```
Comment (DzyubSpirit) [inline bst.js]: Some logic here and in the recursion is duplicated os it's possible to rewrite it like that:
```suggestion
var isValidBST = root => recursion(root, -Infinity, Infinity);
```

**Review by DzyubSpirit — CHANGES_REQUESTED** [review]

File: bst.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var recursion = function(root, min, max) {
+    if (!root) return true;
+    return root.val < max && root.val > min
```
Comment (DzyubSpirit) [inline bst.js]: A ternary operator can be replaced with `&&` here

**Review by DzyubSpirit — CHANGES_REQUESTED** [review]

File: bst.js (hunk 1 above)
Comment (JuliaGerasymenko) [reply bst.js]: [0], [1] - edge cases

Comment (DzyubSpirit) [reply bst.js]: What do you mean?

File: bst.js (hunk 3 above)
Comment (DzyubSpirit) [reply bst.js]: I meant something different ;)
You can write this without ternary operators, using just `&&` operator.

**Review by DzyubSpirit — APPROVED** [review]
Commented here:
https://github.com/JuliaGerasymenko/Tasks/pull/8#discussion_r352544730

**Pull request merged**